### ISSUE TITLE
Indentation rule

### ISSUE THREAD

**Issue by xereda** [stylelint/stylelint#3427]
Hi, 
I need help regarding a behavior in a rule. 

We use sass-lint (https://github.com/sasstools/sass-lint) on some projects, but we're migrating to stylelint. In sass-lint, we use the `indentation` rule and it has a slightly different behavior in its validation.

Look at the illustrations:

`indentation with sass-lint`:
![indentation_sass-lint](https://user-images.githubusercontent.com/14337850/41973767-59961f8c-79ec-11e8-8d7a-39952050306b.gif)

`indentation with stylelint`:
![indentation_stylelint](https://user-images.githubusercontent.com/14337850/41973794-6e8c498e-79ec-11e8-85a8-e738bbc64662.gif)

We found that there is a difference in the validation. Sass-lint validates even the beginning of the style block, not allowing indentation in the initial part. If I indent all the contents of the block, the stylelint allows, but the sass-lint does not.

I wanted your opinion on this behavior. Does `sass-lint` validation make sense? Can I simulate the same validation with _stylelint_? 

Thanks in advance for your help,

Jackson

**Comment by ntwb** [comment]
I'm not why sass-lin allows the first line to be indented and that is ok /shrug 

Anyways, the indentation docs [here](https://stylelint.io/user-guide/rules/indentation/) show what stylelint indents.

If you want to indent the closing brace on the last line you can do that per:
> https://stylelint.io/user-guide/rules/indentation/#indentclosingbrace-truefalse

I'm also closing this as the current stylelint behaviour is intended as is.

If you believe it is a bug in stylelint then please create a new issue using the template that includes your stylelint configuration and example code for us to reproduce the issue.

**Comment by hudochenkov** [comment]
Indentation you're asking about should not be a concern of stylelint, because it's HTML indentation. `<style>` tag could be on any indentation level. E. g.:

```html
<!DOCTYPE html>
<html lang="en">
<head>
	<style>
		body {
			background: #fff;
		}
	</style>
</head>
<body>
</body>
</html>
```

Stylelint doesn't care about HTML indentation. It respects whatever indentation HTML has, but it check styles indentation.

**Comment by xereda** [comment]
Note that sass-lint validates the indentation also by considering the left wall of the style block.

![__indentation_sass-lint](https://user-images.githubusercontent.com/14337850/42034321-0aab4e90-7ab6-11e8-8705-8d99cacac936.gif)

**Comment by xereda** [comment]
@hudochenkov But, I'm working, on my projects, with SFC files from Vue.js (* .vue).

Template for vue.js files (.vue):

```js

<template>
  <span class="test">
    { test }
  </span>
</template>

<script>
export default {
  name: 'test',
  data() {
    return {
      test: 'hello world!',
    };
  },
};
</script>

<style lang="scss" scoped>
.test {
  margin: 10px;
}
</style>


```

**Comment by gucong3000** [comment]
I'm [trying to solve this problem](#3557).
I've added an option `baseIndentLevel`, you can set this option to `0` to make the CSS indent level equal to the parent element.

```js
"use strict";
const stylelint = require(".")
stylelint.lint({
  code: `
    <style>
      a {
        display:block;
      }
    </style>`,
  codeFilename: "a.html",
  config: {
    rules: {
      indentation: [
        2,
        {
          baseIndentLevel: 0,
        }
      ]
    },
  },
  fix: true,
}).then(result => console.log(result.output));
```

**Comment by CodeLittlePrince** [comment]
I don't know why, baseIndentLevel doesn't work in my vue project.
![image](https://user-images.githubusercontent.com/17001245/46649476-c6dd6d80-cbcb-11e8-9940-00112f3906c5.png)

**Comment by jeddy3** [comment]
@CodeLittlePrince Please ensure your using the latest version of stylelint (by deleting your `node_modules` folder and any globally installed versions). If that doesn't fix the problem, please create a new issue (using the ["Report a bug"](https://github.com/stylelint/stylelint/issues/new?template=REPORT_A_BUG.md) template) so that we can better help you.
